### PR TITLE
SUBMARINE-1060. Create TensorFlow model serving

### PR DESCRIPTION
**Pull request by KUAN-HSUN-LI**
### What is this PR for?
1. Use `seldon core` to create the TensorFlow model serving.
2. Create model serve rest API.
3. **Inference API is not created in this PR**

### What type of PR is it?
[Feature]

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-1060

### How should this be tested?
The test will be provided after the `minIO` client is created.

### Screenshots (if appropriate)
https://user-images.githubusercontent.com/38066413/141690103-6a3b9c36-4feb-4295-8bcd-3914c9736769.mp4
### Questions:
* Do the license files need updating? No
* Are there breaking changes for older versions? Yes
* Does this need new documentation? Yes

### CODE REVIEW

**Comment by KUAN-HSUN-LI** [comment]
@jeff-901, Can you help me review this PR?
Thanks in advance.

File: submarine-serve/src/test/java/org/apache/submarine/serve/tensorflow/SeldonTFServingTest.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
+// /*
+//  * Licensed to the Apache Software Foundation (ASF) under one
+//  * or more contributor license agreements.  See the NOTICE file
+//  * distributed with this work for additional information
+//  * regarding copyright ownership.  The ASF licenses this file
+//  * to you under the Apache License, Version 2.0 (the
+//  * "License"); you may not use this file except in compliance
+//  * with the License.  You may obtain a copy of the License at
+//  *
+//  *   http://www.apache.org/licenses/LICENSE-2.0
```
Comment (jeff-901) [inline submarine-serve/src/test/java/org/apache/submarine/serve/tensorflow/SeldonTFServingTest.java:10]: Why do you comment all the lines in the file?

File: submarine-serve/src/main/java/org/apache/submarine/serve/utils/SeldonConstants.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  public static final String GROUP = "machinelearning.seldon.io";
+
+  public static final String VERSION = "v1";
+
+  public static final String PLURAL = "seldondeployments";
+
+  public static final String STORAGE_INITIALIZER_IMAGE = "seldonio/rclone-storage-initializer:1.10.0";
+
+  public static final String ENV_SECRET_REF_NAME = "seldon-init-container-secret";
```
Comment (jeff-901) [inline submarine-serve/src/main/java/org/apache/submarine/serve/utils/SeldonConstants.java:34]: The default ENV_SECRET_REF_NAME should be "submarine-serve-secret".

File: submarine-serve/src/main/java/org/apache/submarine/serve/tensorflow/SeldonTFServing.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    V1ObjectMeta v1ObjectMeta = new V1ObjectMeta();
+    v1ObjectMeta.setName(name);
+    v1ObjectMeta.setNamespace(SeldonConstants.DEFAULT_NAMESPACE);
+    setMetadata(v1ObjectMeta);
+
+    setSpec(new SeldonDeploymentSpec(SeldonConstants.SELDON_PROTOCOL));
+
+    SeldonGraph seldonGraph = new SeldonGraph();
+    seldonGraph.setName(name);
+    seldonGraph.setImplementation(SeldonConstants.TFSERVING_IMPLEMENTATION);
```
Comment (jeff-901) [inline submarine-serve/src/main/java/org/apache/submarine/serve/tensorflow/SeldonTFServing.java:38]: Do we need to help the user to setup envSecretRefName?

File: submarine-serve/src/test/java/org/apache/submarine/serve/tensorflow/SeldonTFServingTest.java (hunk 1 above)
Comment (KUAN-HSUN-LI) [reply submarine-serve/src/test/java/org/apache/submarine/serve/tensorflow/SeldonTFServingTest.java:10]: Sorry, I accidentally pushed this file.

File: submarine-serve/src/main/java/org/apache/submarine/serve/tensorflow/SeldonTFServing.java (hunk 3 above)
Comment (KUAN-HSUN-LI) [reply submarine-serve/src/main/java/org/apache/submarine/serve/tensorflow/SeldonTFServing.java:38]: class `SeldonTFServing` inherit the class `SeldonDeployment` which has setup envSecretRefName by default.

**Review by pingsutw — APPROVED** [review]

**Pull request closed without merging**